### PR TITLE
Rename BlockSync -> ChainExchange and tweak parameters

### PR DESCRIPTION
**Pull request by austinabell**
**Summary of changes**
Changes introduced in this pull request:
- I was tweaking parameters to test syncing and decided to just do this now while I'm testing other things
  - Yamux window is now large enough to handle 500 length request (for #801) and switches default to 200 for now as 10 is too low to test well
- Deduplicates RequestResponse codecs using generics

**Reference issue to close (if applicable)**

**Other information and links**

### CODE REVIEW

**Review by dutterbutter — APPROVED** [review]

File: node/forest_libp2p/src/service.rs [hunk 1]
```diff
@@ -364,8 +366,8 @@ pub fn build_transport(local_key: Keypair) -> Boxed<(PeerId, StreamMuxerBox), Er
         .into_authentic(&local_key)
         .expect("Noise key generation failed");
     let mut yamux_config = yamux::Config::default();
-    yamux_config.set_max_buffer_size(1 << 20);
-    yamux_config.set_receive_window(1 << 20);
+    yamux_config.set_max_buffer_size(16 * 1024 * 1024);
+    yamux_config.set_receive_window(16 * 1024 * 1024);
```
Comment (ec2) [inline node/forest_libp2p/src/service.rs:370]: Can you explain a little how these parameters are decided on? You said in the PR description the window size is now large enough, but 1 << 20 is larger than 16 * 1024 * 1024...

**Review by ec2 — APPROVED** [review]
Everything LGTM, just wanna understand the Yamux param changes a little more... The generic codecs is pretty nice <3 very cool

File: node/forest_libp2p/src/service.rs (hunk 1 above)
Comment (austinabell) [reply node/forest_libp2p/src/service.rs:370]: nah it isn't (16_777_216 > 1_048_576) I updated the window size to be equal to the default go one, just to make sure no issues. 1<<20 was an arbitrary number I didn't think we would hit lol

Comment (ec2) [reply node/forest_libp2p/src/service.rs:370]: oh shoot, youre right.... *throws out my math degree*

Comment (austinabell) [reply node/forest_libp2p/src/service.rs:370]: It looks smaller though, right?

**Pull request merged**